src/main/java/com/commercetools/sync/cartdiscounts/CartDiscountSync.java

```diff
@@ -202,11 +200,9 @@ private void handleError(@Nonnull final String errorMessage,
                         .resolveReferences(newCartDiscount)
                         .thenCompose(resolvedDraft -> syncDraft(oldCartDiscountMap, resolvedDraft))
                         .exceptionally(completionException -> {
-                            final ReferenceResolutionException referenceResolutionException =
-                                (ReferenceResolutionException) completionException.getCause();
-                            final String errorMessage = format(FAILED_TO_RESOLVE_REFERENCES, newCartDiscount.getKey(),
-                                             referenceResolutionException.getMessage());
-                            handleError(errorMessage, referenceResolutionException, 1);
+                            final String errorMessage = format(FAILED_TO_RUN_SYNC_DRAFT, newCartDiscount.getKey(),
+                                completionException.getCause().getMessage());
+                            handleError(errorMessage, completionException.getCause(), 1);
```

Comment:
Why do path the getCause() here and not the whole completionException?

Reply:
CompletionException has a behaviour which always wrap the actual cause of exception. Here, if we use the whole completionException, .getMessage() method returns the wrapped exception classname in addition to the original message. It may bring the impact to current library consumer.